packages/snap/src/utils/validation.ts

```diff
@@ -0,0 +1,89 @@
+import { ethers } from 'ethers';
+import type { ChainConfig } from 'src/keyring';
+import { assert, define, object, optional, StructError } from 'superstruct';
+
+import { throwError } from './util';
+
+const EthereumAddress = define(
+  'EthereumAddress',
+  (value) => typeof value === 'string' && ethers.isAddress(value),
+);
```

Comment:
Good use of `define` to great data validators! As a recommendation I suggest to take a look at the module `@metamask/utils` where we already developed a lot of useful validators like [`isValidHexAddress`](https://github.com/MetaMask/utils/blob/e97f2460d23ec9ed0907e5b68af8ecb84b28cb5a/src/hex.ts#L76)

Reply:
Addressed here [60ee50b](https://github.com/MetaMask/snap-account-abstraction-keyring/pull/20/commits/60ee50be21e3477c78a5b456fb7ba3c3ec5ab13c)